### PR TITLE
feat(Cypress and Jest): add jest and cypress test setup

### PR DESCRIPTION
**Pull request by siddanand**
- Add Jest with basic test cases. (https://jestjs.io/docs/tutorial-react)
- Add Jest Coverage (https://jestjs.io/docs/configuration#collectcoverage-boolean)
- Add  Cypress with basic test cases. (Link: https://docs.cypress.io/guides/component-testing/framework-configuration#Next-js-Webpack-4)
- Add Cypress Coverage (https://docs.cypress.io/guides/tooling/code-coverage#Introduction)
- Add nyc coverage threshold setup (https://www.npmjs.com/package/nyc#coverage-thresholds)

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import Head from 'next/head'
```
Comment (chirgjn) [inline pages/index.js:1]: Can you check if these changes are still imposed by `prettier`? We have changed our config so these might need to be reverted.

File: package.json [hunk 2]
```diff
@@ -6,6 +6,12 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
+    "test": "jest",
+    "test:coverage": "jest --coverage",
+    "cypress:open": "cypress open",
+    "cypress:run": "cypress run --spec 'cypress/integration/*.spec.js'",
+    "cypress:coverage": "nyc npm run cypress:run",
+    "coverage:report": "nyc report --check-coverage --statements 100",
```
Comment (chirgjn) [inline package.json:14]:
```suggestion
    "coverage:report": "nyc report --check-coverage --statements 50",
```

**Review by chirgjn — CHANGES_REQUESTED** [review]
Can we remove the Cypress examples, I don't think they're relevant to the project.

File: .babelrc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["next/babel"]
+}
```
Comment (siddanand) [inline .babelrc:3]: For jest and cypress test

File: __tests__/HomeTest.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import Home from "../pages/index";
+
+it("Renders appropriately", () => {
+  render(<Home />);
+  expect(screen.getByRole("heading", { name: "Babitz" })).toBeInTheDocument();
+});
```
Comment (siddanand) [inline __tests__/HomeTest.js:8]: A sample test case

File: cypress.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "baseUrl": "http://localhost:3000",
+  "env": {
+    "codeCoverage": {
+      "url": "/api/__coverage__"
+    }
+  }
+}
```
Comment (siddanand) [inline cypress.json:8]: For cypress coverage

File: cypress/fixtures/example.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
+  "body": "Fixtures are a great way to mock data for responses to routes"
+}
```
Comment (siddanand) [inline cypress/fixtures/example.json:5]: Pre-Installed with cypress

File: cypress/integration/home.spec.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+context("Home", () => {
+  beforeEach(() => {
+    cy.visit("http://localhost:3000");
+  });
+
+  it("should find the title of the homepage", () => {
+    cy.get("h1").contains("Babitz");
+  });
+});
```
Comment (siddanand) [inline cypress/integration/home.spec.js:9]: Sample test case for cypress

File: package.json [hunk 8]
```diff
@@ -6,6 +6,12 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
+    "test": "jest",
+    "test:coverage": "jest --coverage",
+    "cypress:open": "cypress open",
+    "cypress:run": "cypress run --spec 'cypress/integration/*.spec.js'",
+    "cypress:coverage": "nyc npm run cypress:run",
+    "coverage:report": "nyc report --check-coverage --statements 50",
```
Comment (siddanand) [inline package.json:14]: Scripts for cypress and jest

File: cypress.json [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{
+  "baseUrl": "http://localhost:3000",
+  "env": {
+    "codeCoverage": {
+      "url": "/api/__coverage__"
+    }
+  }
```
Comment (chirgjn) [inline cypress.json:7]:
```suggestion
```

File: cypress/fixtures/example.json (hunk 6 above)
Comment (chirgjn) [reply cypress/fixtures/example.json:5]: Let's remove it

File: __mocks__/fileMock.js [hunk 10]
```diff
@@ -0,0 +1 @@
+module.exports = "placeholder-file";
```
Comment (chirgjn) [inline __mocks__/fileMock.js:1]: Can you add some reference link here

File: __mocks__/styleMock.js [hunk 11]
```diff
@@ -0,0 +1 @@
+module.exports = {};
```
Comment (chirgjn) [inline __mocks__/styleMock.js:1]: Can you add some reference link here

**Review by chirgjn — CHANGES_REQUESTED** [review]

File: __mocks__/fileMock.js (hunk 10 above)
Comment (siddanand) [reply __mocks__/fileMock.js:1]: https://circleci.com/blog/next-testing/

**Review by chirgjn — APPROVED** [review]

**Pull request merged**